hazelcast/src/test/java/com/hazelcast/internal/networking/nio/AdvancedNetworkIntegrationTest.java

```diff
@@ -76,6 +78,31 @@ public void testCompleteMultisocketConfig() {
         assertWrongProtocolAlert(MEMCACHE_PORT, Protocols.CLUSTER, "AAA");
     }
 
+    @Test
+    @Category(QuickTest.class)
+    public void testTPCPortsWithAdvancedNetwork() {
+        Config config = smallInstanceConfig();
+        config.getTpcEngineConfig().setEnabled(true).setEventloopCount(4);
+        ServerSocketEndpointConfig clientSSConfig = new ServerSocketEndpointConfig();
+        clientSSConfig.getTpcSocketConfig()
+                .setReceiveBufferSize(1 << 20)
+                .setSendBufferSize(1 << 19)
+                .addPortDefinition("14000-" + (14000 + 4 - 2))
+                .addPortDefinition("15000-16000");
+        config.getAdvancedNetworkConfig().setEnabled(true).setClientEndpointConfig(clientSSConfig);
+
+        HazelcastInstance[] hz = new HazelcastInstance[3];
+        for (int i = 0; i < 3; i++) {
+            hz[i] = newHazelcastInstance(config);
+        }
+
+        for (int i = 0; i < 3; i++) {
+            assertEquals(1 << 20, getNode(hz[i]).getNodeEngine().getTpcServerBootstrap().getClientSocketConfig().getReceiveBufferSize());
```

Comment:
Testing for exact equality will lead to problems. On Linux, the OS can decide the double the send/receive buffer sizes for example.

Reply:
We are testing equality of what tpc server sees from the config, not the actual config buffer size. So no difference should be observed.